httpmw/ip_cognito_auth.go

```diff
@@ -193,15 +192,16 @@ func IpCognitoAuth(ipRange string, svc cognitoidentityprovideriface.CognitoIdent
 			return
 		}
 
-		model, ok := ch.Get(token)
-
-		if v, ok := model.(*CognitoUser); ok {
-			user = v
-		}
-
-		if !ok {
+		uData, err := cache.Get(c, token).Bytes()
+		if err == nil {
```

Comment:
I would suggest simplifying all of this logic a bit, something like:
```go
data, err := cache.Get(c, token).Bytes()

if err != nil && err != redis.Nil {
	httperr.InternalServerError(c, err.Error())
	c.Abort()
	return
}

if err == nil {
	if err := json.Unmarshal(data, user); err != nil {
		httperr.InternalServerError(c, err.Error())
		c.Abort()
		return
	}
}

if err == redis.Nil {
	res, err := svc.GetUser(&cognitoidentityprovider.GetUserInput{AccessToken: &token})

	if err != nil {
		httperr.InternalServerError(c, err.Error())
		c.Abort()
		return
	}

	user.SetUsername(*res.Username)
	data, err := json.Marshal(user)

	if err != nil {
		httperr.InternalServerError(c, err.Error())
		c.Abort()
		return
	}

	cache.Set(c, token, data, expire)
}
```

Reply:
Also I would suggest avoiding logging here as much as possible, instead throw error messages with status code.

Reply:
Makes sense.
Shouldn't we use httperr.Unauthorized here?

```
res, err := svc.GetUser(&cognitoidentityprovider.GetUserInput{AccessToken: &token})

	if err != nil {
		httperr.InternalServerError(c, err.Error())
		c.Abort()
		return
	}

Reply:
Yeah, for sure.